source/windows/iocp/socket.c

```diff
@@ -716,15 +716,17 @@ void s_socket_connection_completion(
         struct aws_socket *socket = socket_args->socket;
         socket->readable_fn = NULL;
         socket->readable_user_data = NULL;
-        socket_args->socket = NULL;
         socket_impl->connect_args = NULL;
 
         if (!status_code) {
             socket_impl->vtable->connection_success(socket);
         } else {
+            socket_args->socket->state = ERRORED;
             int error = s_determine_socket_error(status_code);
             socket_impl->vtable->connection_error(socket, error);
         }
+
+        socket_args->socket = NULL;
```

Comment:
why did this get moved down here?

Reply:
this isn't need at all.

Reply:
oh wait, wrong function, nevermind, this is fine.

Reply:
This was moved because of the above change, which we no longer need. Reverted.